File: Minimum Increment to Make Array Unique.cpp
class Solution {
public:
    int minIncrementForUnique(vector<int>& nums) {
        sort(nums.begin(), nums.end());
        int minn = nums[0], count = 0;
        for (int& num: nums) {
            minn = max(minn, num);
            if (num < minn) {
                count += minn - num;
            }
            minn++;
        }
        return count;
    }
};

File: Minimum Increment to Make Array Unique.py
class Solution:
    def minIncrementForUnique(self, nums: List[int]) -> int:
        nums.sort()
        minn = nums[0]
        count = 0
        for num in nums:
            minn = max(minn, num)
            if num < minn:
                count += minn - num
            minn += 1
        return count
